mm/shm/shmdt.c

```diff
@@ -113,6 +91,16 @@ int shmdt(FAR const void *shmaddr)
       goto errout_with_errno;
     }
 
+  if (!group)
```

Comment:
the shmdt_priv is called from two places. "shmdt" is the normal detach function for user, and there the "group" always exists.

shmdt_priv is also called from "munmap_shm" in shmat.c. From there the group can also be "null" in case unmap is being done at process destruction phase.

Reply:
World is rarely perfect :)

But that's why I made a private function shmdt_priv, so that you don't need to duplicate code for the two cleanup cases 1) normal munmap 2) process exit (where you must still return pages and deallocate kernel memory, but cannot touch group or MMU)